odoorpc/rpc/jsonrpclib.py

```diff
@@ -110,6 +110,7 @@ def __call__(self, url, params=None):
         data_json = json.dumps(data)
         request = Request(url=full_url, data=encode_data(data_json))
         request.add_header('Content-Type', 'application/json')
+        request.add_header('User-Agent', 'Mozilla/5.0 (Windows; U; Windows NT 5.1; en-US; rv:1.9.0.7) Gecko/2009021910 Firefox/3.0.7')
```

Comment:
Sorry, a bit late on this issue. I don't think hardcoding a user-agent is a good fix here (which one should be used? How to maintain that?).
If the reverse proxy in front of Odoo does not allow HTTP requests without `User-Agent`, I would instead use a custom `opener`, like this:
```python
import urllib.request
import odoorpc

opener = urllib.request.build_opener()
opener.addheaders = [('User-agent', 'Mozilla/5.0')]    # Put the User-Agent you want

odoo = odoorpc.ODOO(..., opener=opener)
```
Then all requests generated by OdooRPC will use this User-Agent.